test/common/ssl/connection_impl_test.cc

```diff
@@ -278,80 +279,61 @@ TEST_P(SslConnectionImplTest, SslError) {
 class SslReadBufferLimitTest : public SslCertsTest,
                                public testing::WithParamInterface<Network::Address::IpVersion> {
 public:
-  void readBufferLimitTest(uint32_t read_buffer_limit, uint32_t expected_chunk_size,
-                           uint32_t write_size, uint32_t num_writes, bool reserve_write_space) {
-    Stats::IsolatedStoreImpl stats_store;
-    Event::DispatcherImpl dispatcher;
-    Network::TcpListenSocket socket(Network::Test::getCanonicalLoopbackAddress(GetParam()), true);
-    Network::MockListenerCallbacks listener_callbacks;
-    Network::MockConnectionHandler connection_handler;
-
-    std::string server_ctx_json = R"EOF(
-    {
-      "cert_chain_file": "{{ test_tmpdir }}/unittestcert.pem",
-      "private_key_file": "{{ test_tmpdir }}/unittestkey.pem",
-      "ca_cert_file": "{{ test_rundir }}/test/common/ssl/test_data/ca_cert.pem"
-    }
-    )EOF";
-    Json::ObjectSharedPtr server_ctx_loader = TestEnvironment::jsonLoadFromString(server_ctx_json);
-    ContextConfigImpl server_ctx_config(*server_ctx_loader);
-    Runtime::MockLoader runtime;
-    ContextManagerImpl manager(runtime);
-    ServerContextPtr server_ctx(manager.createSslServerContext(stats_store, server_ctx_config));
-
-    Network::ListenerPtr listener = dispatcher.createSslListener(
-        connection_handler, *server_ctx, socket, listener_callbacks, stats_store,
+  void Initialize(uint32_t read_buffer_limit) {
```

Comment:
nit: initialize (I think part of this change is in the other change also, so might be best to get one of these in first and then rebase/focus on the other one).